docs/source/pipeline.rst

```diff
@@ -0,0 +1,72 @@
+.. _pipeline-parallelism:
+
+Pipeline Parallelism
+====================
+
+Pipeline parallelism was original introduced in the 
+`Gpipe <https://arxiv.org/abs/1811.06965>`__  paper and is an efficient 
+technique to train large models on multiple GPUs. 
+
+.. warning ::
+     Pipeline Parallelism is experimental and subject to change.
+
+Model Parallelism using multiple GPUs
+-------------------------------------
+
+Typically for large models 
+which don't fit on a single GPU, model parallelism is employed where certain 
+parts of the model are placed on different GPUs. Although, if this is done 
+naively for sequential models, the training process suffers from GPU under 
+utilization since only one GPU is active at one time as show in the figure 
+below:
+
+.. figure:: _static/img/pipeline_parallelism/no_pipe.png
+
+   The figure represents a model with 4 layers placed on 4 different GPUs 
+   (vertical axis). The horizontal axis represents training this model through 
+   time demonstrating that only 1 GPU is utilized at a time 
+   (`image source <https://arxiv.org/abs/1811.06965>`__).
+
+Pipelined Execution
+-------------------
+
+To alleviate this problem, pipeline parallelism splits the input minibatch into 
+multiple microbatches and pipelines the execution of these microbatches across 
+multiple GPUs. This is outlined in the figure below:
+
+.. figure:: _static/img/pipeline_parallelism/pipe.png
+
+   The figure represents a model with 4 layers placed on 4 different GPUs 
+   (vertical axis). The horizontal axis represents training this model through 
+   time demonstrating that the GPUs are utilized much more efficiently. 
+   However, there still exists a bubble (as demonstrated in the figure) where 
+   certain GPUs are not utilized.
```

Comment:
Would it be useful to give an approximation of the increase in utilization a user can expect when using Pipe? I guess this varies a lot for different workloads but maybe we can take an example workload?

Reply:
I feel the two figures attached provide the user a good idea how GPUs are utilized in a more efficient manner. I feel its probably better to illustrate the speed up of this technique in a benchmark/tutorial instead of our docs where we are mostly documenting the feature and its APIs.